### PR TITLE
Add Object.is

### PR DESCRIPTION
**Pull request by MaxGraey**
**DONE**

### CODE REVIEW

**Comment by dcodeIO** [comment]
Since the test is relatively simple and doesn't rely on runtime features, it could use an `object.json` specifying `--runtime none`.

File: std/assembly/object.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+          // @ts-ignore: typecast
+          return reinterpret<u64>(value1) == reinterpret<u64>(value2);
+        } else {
+          // @ts-ignore: typecast
+          return reinterpret<u32>(value1) == reinterpret<u32>(value2);
+        }
+      }
+      // NaN !== NaN, but they are identical.
+      // @ts-ignore: typecast
+      return bool(i32(isNaN(value1)) & i32(isNaN(value2)));
```
Comment (dcodeIO) [inline std/assembly/object.ts]: Maybe `(value1 != value1) && (value2 != value2)`.

Comment (MaxGraey) [reply std/assembly/object.ts]: Actually, it was originally from the beginning, but current version is more compact and more effective)

**Comment by MaxGraey** [comment]
Done!

**Pull request merged**